tests/scripts/tests-setup.sh

```diff
@@ -3,12 +3,32 @@
 
 set -x
 
-if [ "$CONTAINER_USED" = "integration-test-snapshot" ]; then
-   export ARCH="--arch=$(uname -m)"
+# Install required dependencies only when asked
+if [ "$INSTALL_DEPS" == "yes" ]; then
+    dnf install \
```

Comment:
Isn't it possible to define those dependencies in the `.packit.yml`? E.g. for `srpm_build_deps`.

Looking at the GH CI, `podman` (python bindings), `pytest` and `pytest-timeout` are installed via pip - I assume the versions installed via `dnf` are better aligned, right?

Reply:
So packit use COPR to build RPMs, so `srpm_build_deps` are just packages, which needs to be installed to the plain system to be able to build SRPM. Once we have SRPM, COPR itself will install build dependencies as described in the spec file.

But integration tests are running on a testing farm and packit just uses testing farm API to run tests, but testing farm is independent to packit. Officially all packages required to run tests within tmt framework should be installed as a part of `prepare` phase of each plan using `packages` keyword: https://tmt.readthedocs.io/en/stable/spec/plans.html#prepare

Unfortunately tmt doesn't recognize properly ubuntu, so using `packages` keyword will break current GH CI, because `dnf install` doesn't work on ubuntu, that's why we install those packages in https://github.com/eclipse-bluechi/bluechi/blob/main/.github/workflows/integration-tests.yml#L49
For now I've used dnf only, because it's much easier that pip, but if we really need newer versions not in Fedora we could switch

Reply:
Ah ok, thanks for the explanation! The packit + testing integration is quite complex